packages/camera-web/src/Camera/hooks/useUserMedia.ts

```diff
@@ -197,16 +206,29 @@ export function useUserMedia(
   const { handleError } = useMonitoring();
   const isActive = useRef(true);
 
+  let cameraPermissionState: PermissionStatus['state'] | null = null;
   useEffect(() => {
     return () => {
       isActive.current = false;
     };
   }, []);
 
-  const handleGetUserMediaError = (err: unknown) => {
+  const handleGetUserMediaError = async (err: unknown) => {
     let type = UserMediaErrorType.OTHER;
     if (err instanceof Error && err.name === 'NotAllowedError') {
       type = UserMediaErrorType.NOT_ALLOWED;
+      if (cameraPermissionState) {
+        switch (cameraPermissionState) {
+          case 'denied':
+            type = UserMediaErrorType.NOT_ALLOWED_WEBPAGE;
+            break;
+          case 'granted':
+            type = UserMediaErrorType.NOT_ALLOWED_BROWSER;
+            break;
+          default:
+            type = UserMediaErrorType.NOT_ALLOWED;
+        }
+      }
```

Comment:
instead of the `type` assignation and the `if` statement, you can directly use the switch :

```typescript
switch (cameraPermissionState) {
  case 'denied':
    type = UserMediaErrorType.NOT_ALLOWED_WEBPAGE;
    break;
  case 'granted':
    type = UserMediaErrorType.NOT_ALLOWED_BROWSER;
    break;
  default:
    type = UserMediaErrorType.NOT_ALLOWED;
}
```

Reply:
now implementation fixed.